src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -57,9 +69,47 @@ public double getAverageScore() {
    * @param topNComments the number of highest magnitudes to retrieve
    */
   public Statistics(List<UserComment> userCommentList, int topNComments) {
+    wordsToIgnore = CommonWordsRetriever.getCommonWords();
     sentimentBucketList = categorizeToBucketList(userCommentList, topNComments);
     averageScore = getAverageValue(userCommentList, "score");
     averageMagnitude = getAverageValue(userCommentList, "Magnitude");
+    wordFrequencyMap = countWordFrequencyMap(userCommentList);
+  }
+
+  /**
+   * Convert given userCommentList into a word map: {word: frequency}
+   *
+   * @param userCommentList a list of userComment with all fields updated
+   * @return wordFrequencyMap to represent each word appearance time
+   */
+  private Map<String, Integer> countWordFrequencyMap(List<UserComment> userCommentList) {
+    // Flatten all user comment message into a list of words
+    Stream<String> allWordStream =
+        // Text extractor removes all HTML tags and returns only the text
+        userCommentList.stream()
+            .map(
+                comment ->
+                    new Source(comment.getCommentMsg())
+                        .getTextExtractor()
+                        .toString()
+                        .replaceAll("[^a-zA-Z0-9\\s]", "")
+                        .split("\\s+"))
+            .map(wordArray -> new ArrayList<>(Arrays.asList(wordArray)))
+            .flatMap(wordList -> wordList.stream())
+            .filter(word -> !wordsToIgnore.contains(word.toLowerCase()));
+    // Group and sum the appearances of each word
+    Map<String, Integer> wordPairMap =
+        allWordStream.collect(
+            Collectors.groupingBy(word -> word, Collectors.summingInt(word -> 1)));
+
+    if (wordPairMap.size() < MINIMUM_WORDMAP_SIZE) {
```

Comment:
Is it worthwhile to have this behaviour defined on the backend? An empty word map is a valid representation (perhaps the video had no comments).

It seems more natural to have this live on the front end. By returning `null` here the front end code will have to be slightly more complex so that it can handle `null` and non-null inputs. 

Alternatively, if we want to enforce the minimum here, it will be more natural to return an empty map instead of `null`.